src/timeline/InputBar.cpp

```diff
@@ -623,6 +626,54 @@ InputBar::confetti(const QString &body, bool rainbowify)
     room->sendMessageEvent(confetti, mtx::events::EventType::RoomMessage);
 }
 
+void
+InputBar::rainfall(const QString &body, bool rainbowify)
+{
+    auto html = utils::markdownToHtml(body, rainbowify);
+
+    mtx::events::msg::Unknown rain;
+    rain.msgtype = "io.element.effect.rainfall";
+    rain.body    = body.trimmed().toStdString();
+
+    if (html != body.trimmed().toHtmlEscaped() &&
+        ChatPage::instance()->userSettings()->markdown()) {
+        nlohmann::json j;
+        j["formatted_body"] = html.toStdString();
+        j["format"]         = "org.matrix.custom.html";
+        rain.content        = j.dump();
+        // Remove markdown links by completer
+        rain.body = replaceMatrixToMarkdownLink(body.trimmed()).toStdString();
+    }
+
+    rain.relations = generateRelations();
+
+    room->sendMessageEvent(rain, mtx::events::EventType::RoomMessage);
+}
+
+void
+InputBar::customMsgtype(const QString &msgtype, const QString &body, bool rainbowify)
```

Comment:
Well, that last parameter is unused :)

Reply:
Right. I am not sure if it makes sense to extend rainbowifying too far, so I haven't implemented that here. What's your take? Implement it or delete it?

Reply:
Delete. If someone wants to send custom message types, they should rainbowify stuff themselves :)